diff --git a/go/client.mustache b/go/client.mustache
--- a/go/client.mustache
+++ b/go/client.mustache
@@ -54,6 +54,13 @@
 {{/operations}}
 {{/apis}}
 {{/apiInfo}}
+	// PATCH seperate API services for each discriminator identity provider
+	LdapIdentityProvidersApi *LdapIdentityProvidersApiService
+	RadiusIdentityProvidersApi *RadiusIdentityProvidersApiService
+	SamlIdentityProvidersApi *SamlIdentityProvidersApiService
+	LocalDatabaseIdentityProvidersApi *LocalDatabaseIdentityProvidersApiService
+	LdapCertificateIdentityProvidersApi *LdapCertificateIdentityProvidersApiService
+	ConnectorIdentityProvidersApi *ConnectorIdentityProvidersApiService
 }
 
 type service struct {
@@ -79,6 +86,14 @@
 {{/operations}}
 {{/apis}}
 {{/apiInfo}}
+	// PATCH manually added to replace IdentityProvidersApiService
+	// since openapi.generator does not play well with discriminator from the open api spec.
+	c.LdapIdentityProvidersApi = (*LdapIdentityProvidersApiService)(&c.common)
+	c.RadiusIdentityProvidersApi = (*RadiusIdentityProvidersApiService)(&c.common)
+	c.SamlIdentityProvidersApi = (*SamlIdentityProvidersApiService)(&c.common)
+	c.LocalDatabaseIdentityProvidersApi = (*LocalDatabaseIdentityProvidersApiService)(&c.common)
+	c.LdapCertificateIdentityProvidersApi = (*LdapCertificateIdentityProvidersApiService)(&c.common)
+	c.ConnectorIdentityProvidersApi = (*ConnectorIdentityProvidersApiService)(&c.common)
 
 	return c
 }
