diff --git a/go/client.mustache b/go/client.mustache
--- a/go/client.mustache
+++ b/go/client.mustache
@@ -35,6 +35,8 @@
 var (
 	jsonCheck = regexp.MustCompile(`(?i:(?:application|text)/(?:vnd\.[^;]+\+)?json)`)
 	xmlCheck  = regexp.MustCompile(`(?i:(?:application|text)/xml)`)
+	redact    = regexp.MustCompile(`(Authorization: Bearer )(.+)`)
+	redactPW  = regexp.MustCompile(`(?:"password":\w?")(.+?)(?:")`)
 	queryParamSplit = regexp.MustCompile(`(^|&)([^&]+)`)
 	queryDescape    = strings.NewReplacer( "%5B", "[", "%5D", "]" )
 )
@@ -253,7 +255,7 @@
 		if err != nil {
 			return nil, err
 		}
-		log.Printf("\n%s\n", string(dump))
+		log.Printf("\n%s\n", redactPW.ReplaceAllString(redact.ReplaceAllString(string(dump), "${1}REDACTED"), "${2}REDACTED"))
 	}
 
 	resp, err := c.cfg.HTTPClient.Do(request)
@@ -266,7 +268,7 @@
 		if err != nil {
 			return resp, err
 		}
-		log.Printf("\n%s\n", string(dump))
+		log.Printf("\n%s\n", redactPW.ReplaceAllString(redact.ReplaceAllString(string(dump), "${1}REDACTED"), "${2}REDACTED"))
 	}
 	return resp, err
 }
